src/components/frame/v5/pages/FundsPage/context/IncomingFundsLoadingContextProvider.tsx

```diff
@@ -6,23 +6,44 @@ import React, {
   type PropsWithChildren,
 } from 'react';
 
+import useFundsStateUpdater, { useRefetchColonyData } from './hooks.ts';
 import { IncomingFundsLoadingContext } from './IncomingFundsLoadingContext.ts';
 
 export const IncomingFundsLoadingContextProvider: FC<PropsWithChildren> = ({
   children,
 }) => {
   const [isAcceptLoading, setIsAcceptLoading] = useState(false);
+  const [isFundsUpdatePending, setIsFundsUpdatePending] = useState(false);
 
   const enableAcceptLoading = useCallback(() => setIsAcceptLoading(true), []);
-  const disableAcceptLoading = useCallback(() => setIsAcceptLoading(false), []);
+  const enableIsFundsUpdatePending = useCallback(
+    () => setIsFundsUpdatePending(true),
+    [],
+  );
+
+  const reset = useCallback(() => {
+    setIsAcceptLoading(false);
+    setIsFundsUpdatePending(false);
+  }, []);
+
+  useFundsStateUpdater(isFundsUpdatePending, reset);
+  useRefetchColonyData(isAcceptLoading, reset);
```

Comment:
Can these two hooks just get the reset function directly from the context, instead of passing it as a parameter?

Reply:
@davecreaser if we try to use context values - `reset` in this case - before the provider has been initialised, I don't think this change will actually work